march_joint_inertia_controller/src/inertia_estimator.cpp

```diff
@@ -0,0 +1,290 @@
+{
+  lambda_ = lambda;
+}
+void InertiaEstimator::setAcc_size(size_t acc_size)
+{
+  acc_size_ = acc_size;
+}
+
+void InertiaEstimator::setNodeHandle(ros::NodeHandle& nh)
+{
+  nh_ = nh;
+}
+
+void InertiaEstimator::configurePublisher(std::string name)
+{
+  std::string publisher_name = "/inertia_publisher/" + name;
+  pub_ = nh_.advertise<std_msgs::Float64>(publisher_name, 100);
+}
+
+void InertiaEstimator::publishInertia()
+{
+  std_msgs::Float64 msg;
+  msg.data = joint_inertia_;
+  pub_.publish(msg);
+}
+
+// Fills the buffers so that non-zero values may be computed by the inertia estimator
+bool InertiaEstimator::fill_buffers(double velocity, double effort, const ros::Duration& period)
+{
+  auto it = velocity_array_.begin();
+
+  it = velocity_array_.begin();
+  it = velocity_array_.insert(it, velocity);
+  velocity_array_.resize(vel_size_);
+
+  // Automatically fills the zero'th position of the acceleration array
+  discrete_speed_derivative(velocity, period);
+  acceleration_array_.resize(acc_size_);
+
+  it = joint_torque_.begin();
+  it = joint_torque_.insert(it, effort);
+  joint_torque_.resize(torque_size_);
+  return false;
+}
+
+void InertiaEstimator::apply_butter()
+{
+  // Dingen doen met de sos filter ofzo
+  // Also doe pushback dingen anders dan werkt dit niet
+  std::vector<double> z = { 0.0, 0.0, 0.0 };
+  auto it = filtered_acceleration_array_.begin();
+  auto x_n = acceleration_array_[0];
+  double x;
+
+  for (unsigned int i = 0; i < (sizeof(sos_) / sizeof(sos_[0])); ++i)
+  {
+    x_n = acceleration_array_[0];
+    x = sos_[i][0] * x_n + z[0];
+    z[0] = sos_[i][1] * x_n - sos_[i][3] * x + z[1];
+    z[1] = sos_[i][2] * x_n - sos_[i][4] * x;
+
+    //    temp *= (acceleration_array_[0] * sos_[i][0] + acceleration_array_[1] * sos_[i][1] +
+    //             acceleration_array_[2] * sos_[i][2]) /
+    //            (acceleration_array_[0] * sos_[i][3] + acceleration_array_[1] * sos_[i][4] +
+    //             acceleration_array_[2] * sos_[i][5]);
+  }
+  it = filtered_acceleration_array_.begin();
+  it = filtered_acceleration_array_.insert(it, x);
+  filtered_acceleration_array_.resize(acc_size_);
+
+  z = { 0.0, 0.0, 0.0 };
+  for (unsigned int i = 0; i < (sizeof(sos_) / sizeof(sos_[0])); ++i)
+  {
+    x_n = joint_torque_[0];
+    x = sos_[i][0] * x_n + z[0];
+    z[0] = sos_[i][1] * x_n - sos_[i][3] * x + z[1];
+    z[1] = sos_[i][2] * x_n - sos_[i][4] * x;
+    //    temp *= (joint_torque_[0] * sos_[i][0] + joint_torque_[1] * sos_[i][1] + joint_torque_[2] * sos_[i][2]) /
+    //            (joint_torque_[0] * sos_[i][3] + joint_torque_[1] * sos_[i][4] + joint_torque_[2] * sos_[i][5]);
+  }
+  it = filtered_joint_torque_.begin();
+  it = filtered_joint_torque_.insert(it, x);
+  filtered_joint_torque_.resize(fil_tor_size_);
+
+  // temp = 1.0;
+}
+
+// Estimate the inertia using the acceleration and torque
+void InertiaEstimator::inertia_estimate()
+{
+  double torque_e;
+  double acc_e;
```

Comment:
You don't have to initialize variables at the start of a method, instead just initaliaze them when you need them